core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,497 @@
+                        n_fi_subsample: int = 5000, prune_threshold: float = None, prune_ratio: float = 0.05, stop_threshold: int = 10,
+                        improvement_threshold: float = 1e-6, max_fits: int = None, **kwargs) -> List[str]:
+        """
+        Performs time-aware recursive feature elimination based on permutation feature importance. While time remains, compute feature importance
+        score for as many features as possible over at least min_fi_samples validation datapoints, discard features whose score is less than or
+        equal to prune_threshold, fit the model on those features and keep the feature subset if its validation score is better, and repeat.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Training data features
+        y : pd.Series
+            Training data labels
+        X_val: pd.DataFrame
+            Validation data features. Can be None.
+        y_val: pd.Series
+            Validation data labels. Can be None.
+        n_train_subsample : int, default 50000
+            If the training dataset has more than this amount of datapoints, sample this many datapoints without replacement and use
+            them as proxy model training data.
+        n_fi_subsample : int, default 5000
+            Sample this many datapoints and shuffle them when computing permutation feature importance. If this number is higher than
+            the number of feature importance datapoints, set n_fi_subsample = number of feature importance datapoints.
+        prune_threshold : float, default None
+            Consider features whose feature importance scores are below this threshold for pruning. If set to None, a synthetic columns
+            whose values come from standard normal distribution will be appended to the original dataset, and any features whose feature
+            importance score is lower than the synthetic feature's score will be considered for pruning.
+        prune_ratio : float, default 0.05
+            Prune up to prune_ratio * number of current features at once whose feature importance scores are below prune_threshold when
+            generating new feature subset candidates. At least one feature is always removed if there are any feature whose importance
+            score is below prune_threshold.
+        stop_threshold : int, default 10
+            If the validation scores of models fit on pruned data do not improve for stop_threshold iterations, end the pruning procedure.
+        improvement_threshold: int, default = 1e-6
+            The newly fitted model's validation score must be >= improvement_threshold * best validation score seen so far for its input
+            feature subset to be considered to be superior to the previous feature subset.
+        max_fits: int, default None
+            If this many models have been fitted during feature pruning, exit the feature pruning loop. Can potentially prevent overfitting.
```

Comment:
would clarify: we re-fit the model using the remaining features after each round of feature-pruning